opportunity.py

```diff
@@ -0,0 +1,173 @@
+                        job._title,
+                        job._location,
+                        job._link,
+                        job._processed,
+                    ),
+                )
+        connection.commit()
+
+
+def list_opportunities(debug: bool, filtered=False) -> List[Opportunity]:
+    """Lists all oppportunities in DB as well as returns them"""
+
+    with utils.instantiate_db_connection() as connection:
+        cursor = connection.cursor()
+
+        if filtered:
+            cursor.execute(f"SELECT * FROM {table_name} WHERE _processed = 0 LIMIT 15")
+        else:
+            cursor.execute(f"SELECT * FROM {table_name}")
+
+        rows = cursor.fetchall()
+
+        return read_all_opportunities(rows, debug)
+
+
+def read_all_opportunities(rows, debug_tool) -> List[Opportunity]:
+    """Helper function designed to return filtered or unfiltered opportunities"""
+    opportunities = []
+
+    for row in rows:
+        _company, _title, _location, _link, _processed = row
+
+        if debug_tool:
+            print("Company:", _company)
+            print("Title:", _title)
+            print("Location:", _location)
+            print("Link:", _link)
+            print("Processed:", _processed)
+            print(" ")
+
+        opportunity = Opportunity(_company, _title, _location, _link, _processed)
+
+        opportunities.append(opportunity)
+
+    return opportunities
+
+
+def update_opportunities_status(data_results):
+    """Updates the status of the jobs to _processed = 1 after it's been sent by the discord bot"""
+
+    with utils.instantiate_db_connection() as connection:
+        cursor = connection.cursor()
+
+        for data_block in data_results:
+            cursor.execute(
+                f"UPDATE {table_name} SET _processed = %s WHERE _company = %s  AND _title = %s  AND _location = %s",
+                (
+                    1,
+                    data_block._company,
+                    data_block._title,
+                    data_block._location,
+                ),
+            )
+
+        connection.commit()
+
+
+def parse_gpt_values(gpt_response) -> List[bool]:
+    """Helper function to parse the gpt response from a str -> List[bool]"""
+
+    return json.loads(gpt_response)
+
+
+def filter_out_opportunities(list_of_opps, gpt_response) -> List[Opportunity]:
+    """Helper function for gpt_job_analyzer() to filter the data"""
+    structured_opps = [
+        opp for opp, response in zip(list_of_opps, gpt_response) if response
+    ]
+
+    print(f"Length after GPT analyzed the jobs: {len(structured_opps)}")
+    return structured_opps
+
+
+def get_parsed_values(prompt) -> List[bool]:
+    """Helper function which returns the parsed values response from GPT"""
+    response = gpt4free.Completion.create(
+        Provider.You,
+        prompt=prompt,
+    )
+
+    parsed_values = parse_gpt_values(response)
+
+    return parsed_values
+
+
+def gpt_job_analyzer(list_of_opps) -> List[Opportunity]:
```

Comment:
Does the introduction of `gpt_job_analyzer` resolve <https://github.com/acmcsufoss/acmOpportunities/issues/5>?